src/Orchard/ContentManagement/DefaultContentManager.cs

```diff
@@ -95,7 +95,7 @@ public class DefaultContentManager : IContentManager {
             var context = new ActivatingContentContext {
                 ContentType = contentTypeDefinition.Name,
                 Definition = contentTypeDefinition,
-                Builder = new ContentItemBuilder(contentTypeDefinition)
+                Builder = new ContentItemBuilder(contentTypeDefinition, _contentDefinitionManager)

```

Comment:
I agree with you, it doesn't look consistent. But here the singularity is that we don't know in advance which part will be welded on the fly through an `ActivatingFilter` that use the `ContentItemBuilder Weld<TPart>()` method...

So, at least, I will do an update to pass here a List  of partDefinition: `contentPartDefinitions`